### PR TITLE
salvage points shop

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
reworked the salvage vendor to be a shop that gives you items in exchange for mining points
2 new things added for it:
- conscription duffel bag, contains everything you need to draft a tider
- gar mesons, mesons in the shape of gar glasses

moved old gear like pka and pickaxe into the lockers so you don't have to buy them
also gave salv mesons roundstart, they had welders but no eye protection very nice

more stuff like shelter capsules later

## Why / Balance
everything in it is tg parity with the following additions:
- seismic charge for 250
- grappling gun for 300
- the slower mining hardsuit for 3k

reward people for mining and they will probably mine harder

## Technical details
- added ShopVendorComponent, its like VendingMachineComponent but instead of using stock it takes a currency controlled by events
- added PointsVendorComponent, makes a shop vendor use mining points
- added ShopInventoryPrototype to store listings

because VendingMachine is god component a bunch of stuff had to be duplicated which is great

## Media

get in were going to lavaland

too poor to get crusher its over

## Requirements
- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes
removed and migrated away salvage vendor restocks, since it is no longer a vending machine

**Changelog**
:cl:
- add: The Salvage Vendor now lets you buy cool things for mining points.
- tweak: The old salvage gear now comes in the lockers.

### CODE REVIEW

**Comment by Avalon-Proto** [comment]
Cannot wait to see this active in game. Though possibly here's a few suggestions for extra bits to stick in there
- MREs, as a basic food option
- Emergency Pens, to stabilize the stupid salvager who stepped into lava
- Standard Mining suits. Cause if you are mining you aren't as likely to find em

**Comment by deltanedas** [comment]
tg has survival pens which ill port later (epipen does next to nothing if you arent crit), but ill add funny suits too

**Comment by Lyndomen** [comment]
Direction review 24 hours

**Comment by ThataKat** [comment]
Okie dokie, we'd like to do a test merge on this (and the voucher PR), but can we wait until after the 26th? There's gonna be events and direction will be a little low on manpower, so it would be tough to monitor.

**Comment by deltanedas** [comment]
sure

File: Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
+using Content.Client.UserInterface.Controls;
+using Content.Shared.DeltaV.VendingMachines;
+using Content.Shared.Stacks;
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
```
Comment (MilonPL) [inline Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs:9]: unused

File: Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        _vendor = _entMan.System<ShopVendorSystem>();
+
+        VendingContents.SearchBar = SearchBar;
+        VendingContents.DataFilterCondition += DataFilterCondition;
+        VendingContents.GenerateItem += GenerateButton;
+        VendingContents.ItemKeyBindDown += (args, data) => OnItemSelected?.Invoke(((ShopVendorListingData) data).Index);
```
Comment (MilonPL) [inline Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs:46]: you can discard args

File: Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        }
+    }
+
+    private void UpdateBalance()
+    {
+        if (_player.LocalEntity is {} user)
+            UpdateBalance(_vendor.GetBalance(_owner, user));
+    }
+
+    private bool DataFilterCondition(string filter, ListData data)
```
Comment (MilonPL) [inline Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs:87]: is there a reason why this cant be static

File: Content.Shared/DeltaV/VendingMachines/ShopVendorComponent.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    /// How long to wait before another item can be bought
+    /// </summary>
+    [DataField]
+    public TimeSpan EjectDelay = TimeSpan.FromSeconds(1.2);
+
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))]
+    public TimeSpan NextDeny;
+
+    [DataField(customTypeSerializer: typeof(TimeOffsetSerializer))]
+    public TimeSpan NextEject;
```
Comment (MilonPL) [inline Content.Shared/DeltaV/VendingMachines/ShopVendorComponent.cs:51]: real

File: Content.Shared/DeltaV/VendingMachines/ShopVendorComponent.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+namespace Content.Shared.DeltaV.VendingMachines;
+
+/// <summary>
+/// A vending machine that sells items for a currency controlled by events.
+/// Does not need restocking.
+/// Another component must handle <see cref="ShopVendorBalanceEvent"/> and <see cref="ShopVendorPurchaseEvent"/> to work.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedShopVendorSystem))]
+[AutoGenerateComponentState, AutoGenerateComponentPause]
```
Comment (MilonPL) [inline Content.Shared/DeltaV/VendingMachines/ShopVendorComponent.cs:14]: \>adds the AutoGenerateComponentPause attribute
\>doesnt mark any fields as AutoPaused

File: Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs (hunk 3 above)
Comment (deltanedas) [reply Content.Client/DeltaV/VendingMachines/UI/ShopVendorWindow.xaml.cs:87]: copypasta

**Comment by ThataKat** [comment]
Thanks for the patience, feel free to go ahead and merge this if it's ready. We'll keep a close eye on it in game and see how people feel

**Comment by deltanedas** [comment]
the draft is coming

**Pull request merged**